py_ecc/bls/api.py

```diff
@@ -52,17 +49,18 @@ def privtopub(k: int) -> BLSPubkey:
 
 def verify(message_hash: Hash32,
            pubkey: BLSPubkey,
-           signature: BLSSignature,
-           domain: Domain) -> bool:
+           signature: BLSSignature) -> bool:
+    signature_point = signature_to_G2(signature)
+    if not is_inf(multiply(signature_point, G2_COFACTOR)):
```

Comment:
It should be `curve_order` from https://github.com/ethereum/py_ecc/blob/c74aca90b7905a1ec9dd1454dc992d5c6f4f28b0/py_ecc/optimized_bls12_381/optimized_curve.py#L20 instead of `G2_COFACTOR`.
```suggestion
    if not is_inf(multiply(signature_point, curve_order)):
```

Reply:
Ooh Yes good pickup!!